src/provider/model/Storage.ts

```diff
@@ -1,13 +1,95 @@
 import localforage from 'localforage';
+import {injectable} from 'inversify';
 
-export const settingsStorage = localforage.createInstance({
-    driver: localforage.INDEXEDDB,
-    name: 'notifications',
-    storeName: 'settings'
-});
-
-export const notificationStorage = localforage.createInstance({
-    driver: localforage.INDEXEDDB,
-    name: 'notifications',
-    storeName: 'notifications'
-});
+import {AsyncInit} from '../controller/AsyncInit';
+
+/**
+ * List of available storages.
+ */
+export const enum StorageMap {
+    NOTIFICATIONS = 'notifications',
+    SETTINGS = 'settings'
+}
+
+@injectable()
+export class Storage extends AsyncInit {
+    /**
+     * This value should be updated any time changes are made to the notification shape.
+     * Must remain in number format.  LocalForage does not support semver.
+     */
+    public static readonly DATABASE_VERSION: number = 1.0;
```

Comment:
Since we don't use this for semver, would we expect/want this to just be an increasing integer? Suggest just go with `1` here to emphasise

Reply:
I thought to make it decimal to be in a sort of `major.minor` format